File: Domain/Job.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
//using System.ComponentModel.DataAnnotations;

namespace Domain
{
    public class Job
    {
        public Guid Id { get; set; }

      //  [Required]

        public string Title { get; set; }

        public DateTime Date {get; set; }

        public string Description { get; set; }

        public string Category { get; set; }

        public string City { get; set; }
        public bool IsCancelled { get; set; }
        public ICollection<JobAttendee> Attendees { get; set; } = new List<JobAttendee>();
        public ICollection<Comment> Comments { get; set; } = new List<Comment>();

    }
}

File: client-app/src/app/models/job.ts
import { Profile } from "./profile";

export interface Job {
    id: string;
    title: string;
    description: string;
    date: Date | null;
    category: string;
    city: string;
    postUsername: string;
    isCancelled: boolean;
    isGoing: boolean;
    isPost: boolean;
    post?: Profile;
    attendees: Profile []
}

export class Job implements Job{
    constructor(init?: JobFormValues){
        Object.assign(this, init);
    }
}

export class JobFormValues {
    id?: string = undefined;
    title: string = '';
    category: string = '';
    description: string = '';
    date: Date | null = null;
    city: string = '';

    constructor(job?: JobFormValues){
        if(job){
            this.id = job.id;
            this.title = job.title;
            this.category = job.category;
            this.description = job.description;
            this.date = job.date;
            this.city = job.city;
        }
    }
}